### PR TITLE
gestalt-cli migration options

### PR DESCRIPTION
**Pull request by slightlytyler**
PR for options discussed in #11

### CODE REVIEW

**Comment by slightlytyler** [comment]
I added an `importGlob` function that might be better placed with `importAll` in it's repo or a separate repo. It's behavior (besides using glob rather than reading from a directory) is a bit different in that it aggregates all exports from the files, not just the defaults

**Comment by slightlytyler** [comment]
Need to update the documentation before this can be merged

**Comment by charlieschwabacher** [comment]
I see what you mean now - guess it's just easier to read the code 😀...

This is great - It definitely makes sense to use a glob to allow you to have some extra structure in the mutations directory. Thank you!

I have a few comments on code style just to keep things consistent - I'll leave them on the diff.

File: packages/gestalt-cli/src/index.js [hunk 1]
```diff
@@ -30,9 +30,11 @@ commander
     'update schema.json and create a database migration based on changes to ' +
     'schema.graphql in a Gestalt project directory.'
   )
-  .option('-u', '--url <url>', 'the url to your local database')
-  .action(async () => {
-    await migrate();
+  .option('-d, --directory <path>', 'the directory of your mutation files')
+  .option('-g, --glob <pattern>', 'a glob pattern for selecting mutation files')
```
Comment (charlieschwabacher) [inline packages/gestalt-cli/src/index.js]: What do you think about naming this `--mutations-glob` and `--mutations-directory`?  I'm just thinking if you are running `migrate`, its not totally clear that 'directory' would refer to mutations.

File: packages/gestalt-cli/src/migrate.js [hunk 2]
```diff
@@ -82,19 +87,27 @@ async function updateJSONSchema(
...
   localPackage: Object,
   schemaText: string,
+  {url}
 ): Promise {
-  const {databaseURL} = await get({
-    name: 'databaseURL',
-    message: 'what is the url to your database?',
-    default: `postgres://localhost/${snake(localPackage.name)}`,
-  });
+  let databa;
```
Comment (charlieschwabacher) [inline packages/gestalt-cli/src/migrate.js]: Can you leave this as `databaseURL` - I just prefer more descriptive vs abbreviated variable names?

File: packages/gestalt-cli/src/migrate.js [hunk 3]
```diff
@@ -82,19 +87,27 @@ async function updateJSONSchema(
...
+
+  if (url) {
+    databa = url;
+  } else {
+    const prompt = await get({
+      name: 'databa',
+      message: 'what is th to your database?',
```
Comment (charlieschwabacher) [inline packages/gestalt-cli/src/migrate.js]: lost a few characters here

File: packages/gestalt-cli/src/migrate.js [hunk 4]
```diff
@@ -58,9 +60,12 @@ export default async function migrate() {
 
 async function updateJSONSchema(
   localPackage: Object,
-  schemaText: string
+  schemaText: string,
+  {directory, glob}
```
Comment (charlieschwabacher) [inline packages/gestalt-cli/src/migrate.js]: can you add flow types on these?

File: packages/gestalt-cli/src/migrate.js [hunk 5]
```diff
@@ -19,7 +20,8 @@ import {invariant} from 'gestalt-utils';
 import {get} from './cli';
 import 'babel-register';
 
-export default async function migrate() {
+export default async function migrate({directory = 'mutations', glob, url}: {directory: string, glob: string, url: string}) {
```
Comment (charlieschwabacher) [inline packages/gestalt-cli/src/migrate.js]: can you break this line up? - I forgot to add something to 'contributing.md', but the rest of the code uses an 80 column ruler and splits using this format:

```
export default async function migrate({
  directory = 'mutations',
  glob,
  url,
}: {
  directory: string,
  glob: string,
  url: string
}): Promise {
```

**Comment by slightlytyler** [comment]
Is `--url` setup as expected? If provided I set it to skip the prompt. 
Also, is `import-glob` ok where it's at? 
Think I got everything else, might need your help editing the docs as I'm not really sure what style you want to explain the CLI options

**Comment by charlieschwabacher** [comment]
Awesome, thanks!

Re: url - yeah that is great.. it was my mistake that it didn't do use the option in the first place.

Re: import glob - yeah, I can see that being broadly useful.. maybe it should be it's own module?  If you don't want to bother publishing it, I don't mind it where it is either though.

Re: docs - maybe just add the available flags to the bullets below `gestalt migrate` here: https://github.com/charlieschwabacher/gestalt/blob/master/packages/gestalt-cli/README.md

**Comment by slightlytyler** [comment]
@charlieschwabacher I'll leave `import-glob` where it's at, PR is ready

**Comment by charlieschwabacher** [comment]
💸

**Pull request merged**